### DISCUSSION TITLE
fastify-zod

### DISCUSSION THREAD

**Discussion by elierotenberg** [fastify/fastify#3413] [Show and tell]
Hi!
I've just created a library, [`fastify-zod`](https://github.com/elierotenberg/fastify-zod) to facilitate using [`zod`](https://github.com/colinhacks/zod) and `fastify` together, including generating a clean OpenAPI spec using `fastify-swagger`, and use this spec to generate a human-friendly OpenAPI client using the `typescript-fetch` generator.

I would be very happy to get feedback and if you think it's good, add it to the Ecosystem section of the documentation.

What do you think?

N.B: I also started a [similar discussion in the zod respository](https://github.com/colinhacks/zod/discussions/735).

**Comment by RafaelGSS** [comment]
Hi!

Looks good! However, I strongly suggest you add tests.

**Reply by elierotenberg** [reply]
They don't cover all the use cases but there are some [already there](https://github.com/elierotenberg/fastify-zod/tree/master/fastify-zod/src/__tests__).

**Reply by RafaelGSS** [reply]
Go ahead and create the PR for the ecosystem. If someone has a suggestion, they will add it there.
